### DISCUSSION TITLE
About XRArkitBlend shape

### DISCUSSION THREAD

**Discussion by karim-moon** [pixiv/three-vrm#1263] [Help] [answered]
Hi there,

I do really appreciate about this fancy open source project.
I solve many problem about vrm rendering on three.js environment using this package.

Currently, I want to use [XRArkit expression](https://docs.unity3d.com/Packages/com.unity.xr.arkit@1.0/api/UnityEngine.XR.ARKit.XRArkitBlendShapeLocation.html) (not sure it's a right name 😂) to morph vrm's head expression.

I've searched documentation about this, but it seems like three-vrm doesn't support this keys.
Is there a documentation about this? or if it isn't, Have you guys this on road map?

Thanks

**Comment by ke456-png** [comment, marked as answer]
Sorry for the late reply.

There is a mechanism in vrm that allows the 3D model creator to add an extended facial expression as a [Custom Expressions.](https://github.com/vrm-c/vrm-specification/blob/master/specification/VRMC_vrm-1.0/expressions.md#custom-expressions)

It seems that `XRArkit expression` is called `パーフェクトシンク` in the user community, and a corresponding 3D model has been created.

**Reply by karim-moon** [reply]
Thank you for the answer.

We solved this problem last week by adding custom keys as manually.

The term you let me know is also very helpful.

Thanks
